words/core.asm

```diff
@@ -1917,19 +1582,25 @@ question_do_runtime:
         ; """This is called (?DO) in some Forths. See the explanation at
         ; do_runtime for the background on this design
         ; """
-                ; see if TOS and NOS are equal. Change this to assembler
-                ; for speed
-                jsr xt_two_dup          ; ( n1 n2 n1 n2 )
-                jsr xt_equal            ; ( -- n1 n2 f )
-
-                jsr zero_test_runtime   ; consume f, setting Z
-                beq question_do_runtime_end+2
-                inx                     ; drop loop limits
+                ; if TOS == NOS we skip the loop and drop the limits
+                lda 0,x
+                cmp 2,x
+                bne _begin
+                lda 1,x
+                cmp 3,x
+                bne _begin
+                inx                     ; drop loop limits and skip
                 inx
                 inx
                 inx
-                .byte OpJMP             ; jmp
-question_do_runtime_end:
+question_do_runtime_size = * - question_do_runtime
+                ; for native compilation we stop here and ?DO will tack on a JMP <skip-addr>
+                ; for subroutine compile we set up A for zbranch_runtime
+                lda #0
+                .byte $2c               ; BIT llhh to hide the lda #1
```

Comment:
this is the traditional way to hide the next two bytes but will cause a read to $1a0.   i think that's always safe in the stack page?  another option would be a two-byte nop like $dc.